datafusion/sqllogictest/test_files/string_view.slt

```diff
@@ -261,7 +261,7 @@ explain SELECT column1_utf8 from test where column1_utf8view = 'Andrew';
 ----
 logical_plan
 01)Projection: test.column1_utf8
-02)--Filter: test.column1_utf8view = Utf8View("Andrew")
+02)--Filter: test.column1_utf8view = CAST(Utf8("Andrew") AS Utf8View)
```

Comment:
Casts are not simplified to retain the type information to fix the `optimize_projection` issue above.